torch_frame/data/multi_embedding_tensor.py

```diff
@@ -0,0 +1,166 @@
+import copy
+from typing import Any, Callable, Sequence, Tuple
+
+import torch
+from torch import Tensor
+
+
+class MultiEmbeddingTensor:
+    r"""A PyTorch tensor-based data structure that stores
+    :obj:`[num_rows, num_cols, *]`, where the size of last dimension can be
+    different for different column.
+
+    Args:
+        num_rows (int): Number of rows.
+        num_cols (int): Number of columns.
+        values (torch.Tensor): The values :class:`torch.Tensor` of size
+            :obj:`[num_rows, dim1+dim2+...+dimN]`.
+        offset (torch.Tensor): The offset :class:`torch.Tensor` of size
+            :obj:`[num_cols+1,]`.
+    """
+    def __init__(
```

Comment:
I feel a lot of stuff here can be re-used from `MultiNestedTensor`. Does it make sense to create a base class here?

Reply:
Yes, I think apply(), device() logics can definitely be shared. getitem, index logics will be specially implemented for this `MultiNestedTensor`